command/plugin_deregister_test.go

```diff
@@ -131,6 +132,55 @@ func TestPluginDeregisterCommand_Run(t *testing.T) {
 		}
 	})
 
+	t.Run("integration with version", func(t *testing.T) {
+		t.Parallel()
+
+		pluginDir, cleanup := vault.MakeTestPluginDir(t)
+		defer cleanup(t)
+
+		client, _, closer := testVaultServerPluginDir(t, pluginDir)
+		defer closer()
+
+		pluginName := "my-plugin"
+		testPluginCreateAndRegisterVersioned(t, client, pluginDir, pluginName, consts.PluginTypeCredential)
+
+		ui, cmd := testPluginDeregisterCommand(t)
+		cmd.client = client
+
+		code := cmd.Run([]string{
+			consts.PluginTypeCredential.String(),
+			pluginName,
```

Comment:
I don't think this should work without a version specified. If no version is specified, it should try to deregister an unversioned plugin of this type and name, and otherwise do nothing.

Reply:
Oh, good catch. I'll try to fix this.

Reply:
I figured out what was happening. This was working as expected, but the list output didn't have the versioned plugins, so it appeared to succeed.

I added another test to ensure this is working correctly with and without versions, and modified the list plugins output to include the details we need.